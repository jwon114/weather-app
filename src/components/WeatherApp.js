import React from 'react';
import API from '../utils/api';
import Header from './Header';
import ProgressBar from './ProgressBar';
import FiveDayForecast from './FiveDayForecast';

export default class WeatherApp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentTime: undefined,
      currentLocation: 'london',
      currentTemperature: undefined,
      startTimer: false,
      forecastData: []
    };
  }

  componentDidMount() {
    this.getWeatherData();
  }

  getWeatherData = () => {
    API.fetchData()
    .then(res => {
      console.log(res, 'weatherData')
      this.setState(() => ({
        currentTime: this.getCurrentTime(),
        currentTemperature: Math.floor(res.data.current.temp),
        startTimer: true,
        forecastData: res.data.daily.slice(1, 6)
      }));
    })
    .catch(err => {
      console.log(err);
    })
  }

  // bonus
  // use localStorage to keep latest copy of temperature data in case API call fails

  getCurrentTime = () => {
    const dateTimeNow = new Date();
    return `${dateTimeNow.getUTCHours()}:${dateTimeNow.getUTCMinutes().toString().padStart(2, 0)} GMT`;
  }

  handleTimerFinish = () => {
    console.log('timer finished')
    // stop timer
    this.setState(() => ({ startTimer: false }));

    // reset start timer again
    // this.getWeatherData();
  }

  // Todo:
  // day and temperature alignment
  // progress bar smooth animation
  // animation when data updates
  // 

  
  render() {
    console.log(this.state, 'this.state in render')
    const { currentTime, currentLocation, currentTemperature, startTimer, forecastData } = this.state;

    return (
      <div className="WeatherApp">
        <Header 
          location={currentLocation}
          time={currentTime}
          temperature={currentTemperature}
        />
        <ProgressBar
          startTimer={startTimer}
          initialTime={5}
          handleTimerFinished={this.handleTimerFinish}
        />
        <FiveDayForecast 
          data={forecastData}
        />
      </div>
    );
  }
}