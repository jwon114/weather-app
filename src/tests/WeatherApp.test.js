import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import WeatherApp from '../components/WeatherApp';

test('should render WeatherApp correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<WeatherApp />);
  
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});