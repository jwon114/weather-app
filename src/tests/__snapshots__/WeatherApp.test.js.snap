// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WeatherApp correctly 1`] = `
<div
  className="WeatherApp"
>
  <Header
    location="london"
    time="16:17 GMT"
  />
  <ProgressBar
    handleTimerFinished={[Function]}
    initialTime={5}
    startTimer={false}
  />
  <FiveDayForecast
    data={Array []}
  />
</div>
`;
